proxy/connect/connect.go

```diff
@@ -130,12 +136,31 @@ sendLoop:
 	for {
 		select {
 		case request := <-d.sendChannel:
+			select {
+			case <-d.statsTicker.C:
```

Comment:
Can this happen out-of-band of the critical path here? For example, in a separate goroutine?

Reply:
I do need the `request` value to calculate the time difference, so that part has to happen in-band. Putting this call in a goroutine if `shouldCalculateMetrics` is true doesn't have significant performance impact, but seems unnecessarily complex. Wdyt?